### PR TITLE
Add types/jsonwebtokens-promisified

### PR DESCRIPTION
**Pull request by aneilbaboo**
# Overview

The `jsonwebtoken-promisified` npm module adds a couple of methods: `verifySync` and `decodeSync`. This new declaration covers the methods of `jsonwebtoken` and adds these extra promise-returning methods.

# Checklist
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1858,6 +1858,7 @@
 /types/jsonstream/                              @Bartvds
 /types/jsontoxml/                               @benstevens48
 /types/jsonwebtoken/                            @SomaticIT @danielheim @brikou
+/types/jsonwebtoken-promisified                 @aneilbaboo
 /types/jspdf/                                   @amberjs
```
Comment (RyanCavanaugh) [inline .github/CODEOWNERS]: Not a problem but FYI you don't need to modify this file yourself - it is supposed to be auto-generated

File: types/jsonwebtoken-promisified/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (RyanCavanaugh) [inline types/jsonwebtoken-promisified/tslint.json]: I need some justification for why you've disabled all the lint rules. We have them all in place for a reason.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/jsonwebtoken-promisified/tslint.json (hunk 2 above)
Comment (aneilbaboo) [reply types/jsonwebtoken-promisified/tslint.json]: I see what you mean. I copied the tsconfig from types/jsonwebtoken.

Comment (aneilbaboo) [reply types/jsonwebtoken-promisified/tslint.json]: Thanks for catching that. The linting caught a lot of stupid stuff.  I'm fixing it.

I looked at some of the nearby folders, and noticed ~50% of them have the same tsconfig. I suspect DefinitelyTyped is littered with this bad pattern. 

E.g,: jsonnet, jsonpath jsonstream, jspdf, jsrender

Comment (aneilbaboo) [reply types/jsonwebtoken-promisified/tslint.json]: Ah, I see what's happening. The javascript has function signatures that aren't easily matched by typescript rules. I believe the underlying library must be doing runtime inspection of arguments. So enabling the linting rules creates a bunch of errors.

**Comment by aneilbaboo** [comment]
@RyanCavanaugh, Removed the tslint ignores and resulting issues

**Pull request merged**